ttnn/cpp/ttnn/events.hpp

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <memory>
+
+#include "tt_metal/impl/device/device_mesh.hpp"
+
+namespace ttnn {
+
+namespace events {
+
+struct MultiDeviceEvent
+{
+    MultiDeviceEvent(DeviceMesh* device_mesh) {
+        auto& devices = device_mesh->mesh_devices;
+        this->events = std::vector<std::shared_ptr<Event>>(devices.size());
+        for (int event_idx = 0; event_idx < devices.size(); event_idx++) {
+            this->events[event_idx] = std::make_shared<Event>();
+            this->events[event_idx]->device = devices.at(event_idx).second;
```

Comment:
dont need to use .at. 
If devices is the map (I didn't check it) so it will simply be devices[event_idx].

Reply:
done